packages/block-editor/src/hooks/block-style-variation.js

```diff
@@ -59,6 +60,85 @@ function getVariationNameFromClass( className, registeredStyles = [] ) {
 	return null;
 }
 
+/**
+ * This hook is used to generate new block style variation overrides
+ * based on an incoming theme config. If a matching style is found in the config,
+ * a new override is created and returned.
+ *
+ * @param {Object} config A global styles object, containing settings and styles.
+ * @return {Array} An array of new block variation overrides.
+ */
+export function useUpdateBlockVariationOverridesWithConfig( config ) {
```

Comment:
Actually, what I'm thinking would be better than all this is for `useBlockStyleVariation()` to have access to a mergedConfig that contains the current revision. 

https://github.com/WordPress/gutenberg/blob/8ba1ae20c98258bfc9ff13b3a19cee7879025334/packages/block-editor/src/hooks/block-style-variation.js#L142-L153

Maybe it could be return value from `useContext( GlobalStylesContext );`, e.g., 

```js
const { merged: mergedConfig, currentRevisionConfig } = useContext( GlobalStylesContext );
```

I'm just riffing here. Not certain if it'd work.

However, I think this scenario should be taken into account for:

- https://github.com/WordPress/gutenberg/issues/62216

One of the suggestions is to have the site editor recognise global styles as a valid entity, as it does for templates, pages and so on. Global styles could not only be the current mergedConfig, but a revision config. 

If I get time I'll start looking at that.

Reply:
Both options sound worth exploring post 6.6>

```suggestion
export function useUpdateBlockStyleVariationOverridesWithConfig( config ) {
```

I know the name is getting pretty long but I think it helps to be clear this is to do with block style variations, not block variations.